apis/goharbor.io/v1alpha2/component.go

```diff
@@ -0,0 +1,111 @@
+package v1alpha2
+
+import (
+	"encoding/json"
+
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/kustomize/kstatus/status"
+)
+
+type ComponentSpec struct {
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:default=1
+	// Replicas is the number of desired replicas.
+	// This is a pointer to distinguish between explicit zero and unspecified.
+	// More info: https://kubernetes.io/docs/concepts/workloads/controllers/replicationcontroller#what-is-a-replicationcontroller
+	Replicas *int32 `json:"replicas,omitempty" protobuf:"varint,1,opt,name=replicas"`
+
+	// +kubebuilder:validation:Optional
+	// ServiceAccountName is the name of the ServiceAccount to use to run this component.
+	// More info: https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/
+	ServiceAccountName string `json:"serviceAccountName,omitempty" protobuf:"bytes,8,opt,name=serviceAccountName"`
+
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:Enum={"Always","Never","IfNotPresent"}
+	// +kubebuilder:default="IfNotPresent"
+	// Image pull policy.
+	// More info: https://kubernetes.io/docs/concepts/containers/images#updating-images
+	ImagePullPolicy corev1.PullPolicy `json:"imagePullPolicy,omitempty" protobuf:"bytes,14,opt,name=imagePullPolicy,casttype=PullPolicy"`
+
+	// +kubebuilder:validation:Optional
+	// +listType:map
+	// +listMapKey:name
+	ImagePullSecrets []corev1.LocalObjectReference `json:"imagePullSecrets,omitempty" patchStrategy:"merge" patchMergeKey:"name"`
```

Comment:
So, you'd like to allow users to specify different image resource registries for the different component images? I know this is more flexible and more powerful. Why not make it simple to disallow multiple sources of component images. If they need to customize the images, they can use kubectl set image command after installation.